File: src/main/java/de/honoka/bossddmonitor/mapper/JobInfoMapper.java
package de.honoka.bossddmonitor.mapper;

import com.baomidou.mybatisplus.core.conditions.query.LambdaQueryWrapper;
import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import de.honoka.bossddmonitor.entity.JobInfo;
import org.apache.ibatis.annotations.Mapper;

@Mapper
public interface JobInfoMapper extends BaseMapper<JobInfo> {
    
    default Long findIdByPlatformJobId(JobInfo.PlatformEnum platform, String platformJobId) {
        JobInfo result = selectOne(
            new LambdaQueryWrapper<JobInfo>()
                .select(JobInfo::getId)
                .eq(JobInfo::getPlatform, platform)
                .eq(JobInfo::getPlatformJobId, platformJobId)
                .last("limit 1")
        );
        return result != null ? result.getId() : null;
    }
}

File: src/main/java/de/honoka/bossddmonitor/mapper/JobInfoMapper.kt
package de.honoka.bossddmonitor.mapper

import com.baomidou.mybatisplus.core.mapper.BaseMapper
import de.honoka.bossddmonitor.entity.JobInfo
import de.honoka.bossddmonitor.platform.PlatformEnum
import de.honoka.sdk.spring.starter.mybatis.queryChainWrapper
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface JobInfoMapper : BaseMapper<JobInfo> {
    
    fun findByPlatformJobId(platform: PlatformEnum, platformJobId: String): JobInfo? {
        queryChainWrapper().run {
            eq(JobInfo::platform, platform)
            eq(JobInfo::platformJobId, platformJobId)
            last("limit 1")
            return one()
        }
    }
    
    fun getNoRecordsJobInfoList(@Param("userId") userId: Long): List<JobInfo>
}
